src/Database/Orville/Internal/IndexDefinition.hs

```diff
@@ -35,3 +35,21 @@ simpleIndex name tableDef fields concurrently =
 
 indexFieldsBody :: [FieldDefinition] -> String
 indexFieldsBody fields = "(" ++ intercalate "," (map escapedFieldName fields) ++ ")"
+
+newtype IndexName = IndexName String deriving (Eq, Ord)
+
+indexNameToString :: IndexName -> String
+indexNameToString (IndexName i) = i
+
+stringToIndexName :: String -> IndexName
+stringToIndexName name =
+  case safeStringToIndexName name of
```

Comment:
This is fine as it is, but you could use `either` if you want it to be a bit shorter.

``` hs
stringToIndexName = either error id . safeStringToIndexName
```